freeipa/src/main/java/com/sequenceiq/freeipa/service/stack/FreeIpaHealthDetailsService.java

```diff
@@ -0,0 +1,88 @@
+package com.sequenceiq.freeipa.service.stack;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import com.sequenceiq.freeipa.api.v1.freeipa.stack.model.common.DetailedStackStatus;
+import com.sequenceiq.freeipa.api.v1.freeipa.stack.model.common.Status;
+import com.sequenceiq.freeipa.api.v1.freeipa.stack.model.health.HealthDetailsFreeIpaResponse;
+import com.sequenceiq.freeipa.api.v1.freeipa.stack.model.health.NodeHealthDetailsFreeIpaResponse;
+import com.sequenceiq.freeipa.client.FreeIpaClient;
+import com.sequenceiq.freeipa.client.model.RPCMessage;
+import com.sequenceiq.freeipa.client.model.RPCResponse;
+import com.sequenceiq.freeipa.entity.Stack;
+import com.sequenceiq.freeipa.service.TlsSetupService;
+import com.sequenceiq.freeipa.service.freeipa.FreeIpaClientFactory;
+
+@Service
+
+public class FreeIpaHealthDetailsService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(TlsSetupService.class);
+
+    private static final String EXTERNAL_COMMAND_OUTPUT = "ExternalCommandOutput";
+
+    private static final String STATUS_OK = "OK";
+
+    private static final String MESSAGE_UNAVAILABLE = "Message Unavailable";
+
+    private static final Pattern RESULT_PATTERN = Pattern.compile("(ecure port|: TCP) \\([0-9]*\\): (.*)");
+
+    private static final Pattern NEW_NODE_PATTERN = Pattern.compile("Check connection from master to remote replica '(.[^\']*)");
+
+    @Inject
+    private StackService stackService;
+
+    @Inject
+    private FreeIpaClientFactory freeIpaClientFactory;
+
+    public HealthDetailsFreeIpaResponse getHealthDetails(String environmentCrn, String accountId) throws Exception {
+        Stack stack = stackService.getByEnvironmentCrnAndAccountId(environmentCrn, accountId);
+        FreeIpaClient freeIpaClient = freeIpaClientFactory.getFreeIpaClientByAccountAndEnvironment(environmentCrn, accountId);
+
+        RPCResponse<Boolean> rpcResponse = freeIpaClient.serverConnCheck(stack.getResourceCrn(), stack.getResourceCrn());
```

Comment:
FreeIPA doesn't know anything about resource crns, these parameters should be revisited
`cn
required
IPA server hostname`
`remote_cn
required
Remote IPA server hostname`

Reply:
fixed